src/main/java/io/apicurio/datamodels/refs/Reference.java

```diff
@@ -0,0 +1,102 @@
+package io.apicurio.datamodels.refs;
+
+/**
+ * Encapsulates a reference string in OpenAPI/AsyncAPI schema,
+ * for easier manipulation and parsing.
+ *
+ * @author Jakub Senko <jsenko@redhat.com>
+ */
+public class Reference {
+
+
+    private final String ref;
+
+    private String abs;
+    private String rel;
+
+    /**
+     * @param ref Encapsulated reference string. MUST NOT be null.
+     */
+    public Reference(String ref) {
+        if (ref == null)
+            throw new IllegalArgumentException();
```

Comment:
Consider using `Objects#requireNonNull` which would throw an NPE with an optional custom message.

Reply:
I was ready to say that this library has to transpile successfully to TS so `Objects#requireNonNull` might not work.  Then I thought I should actually try it before not using it.  :)  So i tried it and it works!  Way to go, jsweet!

So I'm changing this as suggested.